doc/man7/provider-cipher.pod

```diff
@@ -42,6 +42,10 @@ provider-cipher - The cipher library E<lt>-E<gt> provider functions
  /* Cipher operation parameter descriptors */
  const OSSL_PARAM *OSSL_FUNC_cipher_gettable_ctx_params(void *provctx);
  const OSSL_PARAM *OSSL_FUNC_cipher_settable_ctx_params(void *provctx);
+ const OSSL_PARAM *OSSL_FUNC_cipher_ctx_gettable_params(void *provctx,
+                                                        void *cctx);
+ const OSSL_PARAM *OSSL_FUNC_cipher_ctx_settable_params(void *provctx,
+                                                        void *cctx);
```

Comment:
From a provider author perspective, I find it quite clumsy to have two functions that essentially do the same thing. I'd rather see them merged into one, and being able to vary the response depending on if the second argument is NULL or not.

Reply:
A provider author need only implement the existing function and everything will be happy.  The new function is only required if the parameters can change based on the state of the context.

Not sure if this justifies separate functions or not but it isn't onerous or clumsy for provider authors.

Reply:
I think the one API would be less confusing..

Reply:
Major reworking coming up.  Most of what's been done here is not relevant with this change.

Reply:
I've switched to a single libcrypto ↔ provider API that handles both flavours.  Comments welcomed.